akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -162,23 +165,16 @@ abstract class ActorMaterializer extends Materializer {
    */
   def isShutdown: Boolean
 
-  /**
-   * INTERNAL API: this might become public later
-   */
-  private[akka] def actorOf(context: MaterializationContext, props: Props): ActorRef
+  def materialize[Mat](_runnableGraph: Graph[ClosedShape, Mat],
+                       subflowFuser: GraphInterpreterShell ⇒ ActorRef): Mat
```

Comment:
Yes, I was thinking along those lines as well. I am not sure if this is the best trait decomposition, but we can iterate on this.